src/loggers/parser_logger.cpp

```diff
@@ -1,14 +1,16 @@
 #include "loggers/parser_logger.h"
+
 #include <memory>
-#include "loggers/main_logger.h"
 
 namespace terrier::parser {
 
-std::shared_ptr<spdlog::logger> parser_logger;  // NOLINT
+std::shared_ptr<spdlog::logger> parser_logger = nullptr;  // NOLINT
```

Comment:
I added this initialization and then the nullptr check below to all of the loggers when trying to debug why loggers were getting started and shutdown multiple times in the current system, along with races associated to that. I've cleaned that all up, but should I change the `if` below to just be a `TERRIER_ASSERT` so we don't reintroduce duplicate initialization calls in the future?